stripe/src/main/java/com/stripe/android/cards/RemoteCardAccountRangeSource.kt

```diff
@@ -0,0 +1,30 @@
+package com.stripe.android.cards
+
+import com.stripe.android.ApiRequest
+import com.stripe.android.StripeRepository
+import com.stripe.android.model.CardMetadata
+
+internal class RemoteCardAccountRangeSource(
+    private val stripeRepository: StripeRepository,
+    private val requestOptions: ApiRequest.Options
+) : CardAccountRangeSource {
+
+    override suspend fun getAccountRange(
+        cardNumber: String
+    ): CardMetadata.AccountRange? {
+        return cardNumber
+            .take(BIN_LENGTH)
+            .takeIf {
+                it.length == BIN_LENGTH
+            }?.let { bin ->
+                stripeRepository.getCardMetadata(bin, requestOptions).accountRanges
+                    .firstOrNull { (binRange) ->
```

Comment:
The iOS SDK has some logic handling the case where multiple ranges match. If all the matched ranges have the same brand, they use that brand. Not sure if there's ever a case where the edge service will return ranges with different brands that may both match. cc @csabol-stripe 
https://github.com/stripe/stripe-ios/blob/53d11b48/Stripe/STPCardValidator.m#L225